src/power_process.c

```diff
@@ -667,4 +668,83 @@ void remove_explored_flags_for_power_sight(struct PlayerInfo *player)
         }
     }
 }
+
+void process_timebomb(struct Thing *creatng)
+{
+    SYNCDBG(18,"Starting");
+    if (!creature_affected_by_spell(creatng, SplK_TimeBomb)) {
+        return;
+    }
+    update_creature_speed(creatng);
+    struct CreatureControl* cctrl = creature_control_get_from_thing(creatng);
+    struct Thing* timetng = thing_get(cctrl->timebomb_countdown_id);
+    if (thing_is_invalid(timetng))
+    {
+        if ((cctrl->timebomb_countdown % game.num_fps) == 0)
+        {
+            long time = (cctrl->timebomb_countdown / game.num_fps);
+            timetng = create_price_effect(&creatng->mappos, creatng->owner, time);
+            cctrl->timebomb_countdown_id = timetng->index;
+            thing_play_sample(creatng, 853, NORMAL_PITCH, 0, 3, 0, 2, FULL_LOUDNESS);
+        }
+    }
+    else
+    {
+        if (timetng->health <= 0)
+        {
+            delete_thing_structure(timetng, 0);
+            cctrl->timebomb_countdown_id = 0;
+        }
+        else
+        {
+            struct Coord3d pos;
+            pos.x.val = creatng->mappos.x.val;
+            pos.y.val = creatng->mappos.y.val;
+            pos.z.val = timetng->mappos.z.val;
+            move_thing_in_map(timetng, &pos);
+        }
+    }
+    struct Thing* trgtng = thing_get(cctrl->timebomb_target_id);
+    if (!thing_is_invalid(trgtng))
+    {
+        if ( (creatng->mappos.x.stl.num == trgtng->mappos.x.stl.num) && (creatng->mappos.y.stl.num == trgtng->mappos.y.stl.num) && (creatng->mappos.z.stl.num == trgtng->mappos.z.stl.num) )
+        {
+            timebomb_explode(creatng);
+            return;
+        }
+    }
+    if (cctrl->timebomb_countdown != 0)
+    {
+        cctrl->timebomb_countdown--;
+    }
+    else
+    {
+        timebomb_explode(creatng);
+    }
+}
+
+void timebomb_explode(struct Thing *creatng)
+{
+    if (creature_model_bleeds(creatng->model))
+    {
+        create_effect_around_thing(creatng, TngEff_Blood5);
+    }
+    create_effect_around_thing(creatng, TngEff_Explosion6);
+    struct ShotConfigStats* shotst = get_shot_model_stats(SplK_TimeBomb);
+    HitTargetFlags hit_targets = HitTF_EnemyCreatures|HitTF_AlliedCreatures|HitTF_OwnedCreatures|HitTF_ArmourAffctdCreatrs|
```

Comment:
Why can you not pick a hit_type from 'hit_type_to_hit_targets'? Why would this timebomb explosion do a different damage type compared to say, a meteor explosion?

Reply:
We want all of them, don't we? I don't understand how range is calculated for other explosions.

Reply:
I don't think so no,... spell explosions simply don't hit everything. For example, room flags are never destroyed or moved. Besides, consistency is key.

In fact, we could use SHOT_TIME_BOMB for what happens when the creature is destroyed. Take the hit type from that and other properties we need that are now hardcoded.

Reply:
> So, what should the spell hit?

I would say the same thing as the meteor or grenade explosions (assuming they are the same).

Reply:
How's it now?